Cli/Components/States/Results.razor

```diff
@@ -41,54 +69,127 @@
     protected override void OnInitialized()
     {
         ParseOptsToResultTypes();
-        GetAnalysisResults();
+        UpdatePageCount();
+        GetResultsPage(1);
         base.OnInitialized();
     }
 
-    void MoreAnalysisResults()
+    Dictionary<RESULT_TYPE, int> foundResultTypes = new Dictionary<RESULT_TYPE, int>();
+    public List<CompareResult> analysisResults = new List<CompareResult>();
+    int offset = 0;
+    int currentPage = 1;
+    int pageCount = 0;
+    List<string> pageStates = new List<string>();
+
+    int _maxResults = 100;
+    int MaxResults
     {
-        offset += count;
-        GetAnalysisResults();
+        get { return _maxResults; }
+        set
+        {
+            _maxResults = value;
+            UpdatePageCount();
+            GetResultsPage(1);
+        }
     }
 
-    int offset = 0;
-    int count = 100;
-
+    string _selectedResultType = "FILEMONITOR";
     string SelectedResultType
     {
         get { return _selectedResultType; }
-        set { _selectedResultType = value; offset = 0; GetAnalysisResults(); }
+        set
+        {
+            _selectedResultType = value;
+            UpdatePageCount();
+            GetResultsPage(1);
+        }
     }
 
-    string _selectedResultType = "FILEMONITOR";
+    void GetResultsPage(int selectedPage)
+    {
+        if (pageCount < 1) { return; }
+        SetPageState(selectedPage);
```

Comment:
I don't understand what the page state array and functions are doing.

Reply:
Change this to just set the current page directly.

Reply:
The pageStates list tracks the active page-item within the Bootstrap Pagination nav component. Each page-item's CSS class is either "active" or blank "   "  which is retrieved from pageStates, and the values flow directly into the HTML. SetPageState() is used to manage them accordingly. I'm open to renaming these or reworking the implementation.

Reply:
See later comments. They are not needed.

Reply:
Updated, thanks for the pointers!